Dockerfile

```diff
@@ -1,21 +1,30 @@
-# vim:set ft=dockerfile:
-FROM python:3.8-slim
-MAINTAINER https://github.com/pacificclimate/osprey
-LABEL Description="osprey WPS" Vendor="pacificclimate" Version="0.1.0"
+FROM python:3.8-slim AS builder
 
 ENV PIP_INDEX_URL="https://pypi.pacificclimate.org/simple/"
 
 # Update system
 RUN apt-get update && apt-get install -y \
     build-essential
 
-COPY . /opt/wps
-
 WORKDIR /opt/wps
+COPY ./osprey /opt/wps/osprey
+COPY CHANGES.rst README.rst requirements.txt requirements_dev.txt setup.py ./
```

Comment:
I'm not sure I understand the purpose of these commands. We copy over everything from osprey, then copy things individually again? Can you even use `COPY` in this manner?

Reply:
So the procedure is:
1. Copy the resources required only to install `osprey`, such as `CHANGES.rst README.rst requirements.txt requirements_dev.txt setup.py`, from the repository (no flag required).
2. Run `pip install .` to  install `osprey`
3. Install `gunicorn`
4. `COPY` only necessary artifacts from the intermediate image to the final image (`--from` flag makes it possible). In this case, `gunicorn`, `osprey` and its dependencies are copied.
5. Set `WORKDIR` to the location where `osprey` is installed.
6. Run `gunicorn`

Reply:
@nikola-rados 
>What is the size of the intermediate image that is created?
```
REPOSITORY                  TAG                IMAGE ID       CREATED          SIZE
pcic/osprey                 multi-stage        1b1d586c91e3   20 minutes ago   434MB
<none>                      <none>             a031b2e85c07   22 minutes ago   760MB
```
The `<none>` image is the intermediate image created.

>Does this impact build time at all?
Technically, if the intermediate stage is not modified and the previous intermediate image is not `prune`d, the build step can be skipped. But for this `Dockerfile`, the intermediate stage is likely to be modified most of the time since it copies `osprey` directory, which is frequently changed. I think I can make the caching work better by not using `pip install .` in the intermediate stage.